packages/modular-scripts/src/esbuild-scripts/plugins/rewriteDependenciesPlugin.ts

```diff
@@ -0,0 +1,58 @@
+import * as esbuild from 'esbuild';
+import type { Dependency } from '@schemastore/package';
+
+export function createRewriteDependenciesPlugin(
+  dependencies: Dependency,
+): esbuild.Plugin {
+  const externalCdnTemplate =
+    process.env.EXTERNAL_CDN_TEMPLATE ??
+    'https://cdn.skypack.dev/[name]@[version]';
+
+  const externalBlockList = process.env.EXTERNAL_BLOCK_LIST
+    ? process.env.EXTERNAL_BLOCK_LIST.split(',')
+    : [];
+
+  const importMap: Record<string, string> = Object.entries(dependencies).reduce(
+    (acc, [name, version]) => {
+      if (!externalBlockList.some((filter) => name.startsWith(filter))) {
+        return {
+          ...acc,
+          [name]: externalCdnTemplate
+            .replace('[name]', name)
+            .replace('[version]', version),
+        };
+      }
+      return acc;
+    },
+    {},
+  );
+
+  const dependencyRewritePlugin: esbuild.Plugin = {
+    name: 'dependency-rewrite',
+    setup(build) {
+      build.onResolve({ filter: /.*/, namespace: 'file' }, (args) => {
+        if (args.path in dependencies) {
```

Comment:
How do you handle subpath imports e.g. `react-dom/server` or `lodash/get`?

Reply:
Good point, this needs refining.